packages/core/upload/admin/src/pages/App/MediaLibrary/components/Header.tsx

```diff
@@ -1,30 +1,54 @@
-import React from 'react';
-
 import { useQueryParams, Layouts } from '@strapi/admin/strapi-admin';
 import { Button, Flex, Link } from '@strapi/design-system';
 import { ArrowLeft, Plus } from '@strapi/icons';
-import PropTypes from 'prop-types';
 import { stringify } from 'qs';
 import { useIntl } from 'react-intl';
-import { useLocation } from 'react-router-dom';
+import { useLocation, NavLink } from 'react-router-dom';
+
+import { Breadcrumbs } from '../../../../components/Breadcrumbs/Breadcrumbs';
+import { getTrad } from '../../../../utils/utils';
+
+import type { Folder } from '../../../../../../shared/contracts/folders';
+import type { CrumbDefinition } from '../../../../components/Breadcrumbs/Breadcrumbs';
+
+interface FolderDefinition extends Omit<Folder, 'children' | 'files' | 'parent'> {
+  children: {
+    count: number;
+  };
+  files: {
+    count: number;
+  };
+  parent?: FolderDefinition;
+}
 
-import { Breadcrumbs } from '../../../components/Breadcrumbs';
-import { BreadcrumbsDefinition, FolderDefinition } from '../../../constants';
-import { getTrad } from '../../../utils';
+export interface HeaderProps {
+  breadcrumbs?: Array<CrumbDefinition> | boolean;
```

Comment:
why does the prop accept a boolean?

Reply:
you can see the reason here https://github.com/strapi/strapi/pull/21896#discussion_r1827426665